extensions/amp-mustache/0.1/amp-mustache.js

```diff
@@ -34,13 +34,29 @@ export class AmpMustache extends AMP.BaseTemplate {
   /** @override */
   compileCallback() {
     /** @private @const {string} */
-    this.template_ = this.element./*OK*/innerHTML;
+    const shadowTemplate = this.element.cloneNode(true);
+    this.nestedTemplates_ = {};
+    let index = 0;
+    shadowTemplate.content.querySelectorAll('template').forEach(
+        nestedTemplate => {
+          const nestedTemplateKey = `__AMP_NESTED_TEMPLATE_${index}`;
+          this.nestedTemplates_[nestedTemplateKey] =
+              nestedTemplate./*OK*/outerHTML;
+
+          const nestedTemplateAsVariable = document.createTextNode(
```

Comment:
Prefer `this.element.ownerDocument` over the global `window.document` reference.